### PR TITLE
Remove-DbaServerRoleMember  - new command 

### LINKED ISSUE #7681: Add-DbaServerRoleMember and Remove-DbaServerRoleMember
### Summarize Command's Functionality'

It appears there are commands to add and remove Server roles, add and remove DB roles, and add/remove members from a DB role. However, there are no commands to add/remove members from server roles. I am trying to automate everything in our SQL Server cookbooks (we use Chef) using DBATools and feel this one is critical to get automated.
Thanks!

### Is there a command that is similiar or close to what you are looking for?

Yes

### Technical Details

similar commands:
https://docs.dbatools.io/#Add-DbaDbRoleMember
https://docs.dbatools.io/#Remove-DbaDbRoleMember

### PR DESCRIPTION
**Pull request by MikeyBronowski**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [x] New feature (non-breaking change, adds functionality, fixes #7681 )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Unit test is included
 - [x] Documentation
 - [ ] Build system

### Purpose

The function was missing from the set. See #7681.

### Approach

I've used `Remove-DbaServerRoleMember` as a template and based on that created the `Remove-DbaServerRoleMember`. They are basically identical with 

### Commands to test

`Remove-DbaServerRoleMember -SqlInstance localhost -ServerRole bulkadmin, dbcreator -Login login1`

```
$roles = Get-DbaServerRole -SqlInstance localhost -ServerRole bulkadmin, dbcreator
$roles | Remove-DbaServerRoleMember -Login login1
```

### CODE REVIEW

File: functions/Remove-DbaServerRoleMember.ps1 [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+        Removes all the logins found in C:\logins.txt from mycustomrole custom server-level role on server1.
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    param (
+        [parameter(ValueFromPipeline)]
+        [DbaInstance[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [parameter(ValueFromPipeline)]
+        [string[]]$ServerRole,
```
Comment (andreasjordan) [inline functions/Remove-DbaServerRoleMember.ps1:98]: Sorry, no time for a test right now. So a question: Does this work? Would you pipe in the name of the server role as string and provide all other info as parameters?

File: functions/Remove-DbaServerRoleMember.ps1 [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        [string[]]$Login,
+        [string[]]$Role,
+        [parameter(ValueFromPipeline)]
+        [object[]]$InputObject,
+        [switch]$EnableException
+    )
+
+    begin {
+        if ( (Test-Bound SqlInstance -Not) -and (Test-Bound ServerRole -Not) -and (Test-Bound Login -Not) ) {
+            Stop-Function -Message "You must pipe in a ServerRole, Login, or specify a SqlInstance"
```
Comment (andreasjordan) [inline functions/Remove-DbaServerRoleMember.ps1:108]: Not sure if this message is correct.

Comment (MikeyBronowski) [reply functions/Remove-DbaServerRoleMember.ps1:108]: What's your suggestion? It's a Copy of Add-DbaServerRoleMember so might need changing there too.

File: functions/Remove-DbaServerRoleMember.ps1 (hunk 1 above)
Comment (MikeyBronowski) [reply functions/Remove-DbaServerRoleMember.ps1:98]: It does not work when I pipe in the string

`'customrole' | Remove-DbaServerRoleMember -SqlInstance $s1 -SqlCredential $credential -Login $logins`

it works when piped the object though

```
$r = Get-DbaServerRole -SqlInstance $s1 -SqlCredential $credential -ServerRole $customServerRole
$r | Remove-DbaServerRoleMember -SqlInstance $s1 -SqlCredential $credential -Login $logins
```

If it needs correcting we can make changes to Add-DbaServerRoleMember function as well.

File: functions/Remove-DbaServerRoleMember.ps1 [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        PS C:\> $roles | Remove-DbaServerRoleMember -Login login1
+
+        Removes login1 from the bulkadmin and dbcreator fixed server-level roles on the server localhost.
+
+    .EXAMPLE
+        PS C:\ $logins = Get-Content C:\logins.txt
+        PS C:\ $srvLogins = Get-DbaLogin -SqlInstance server1 -Login $logins
+        PS C:\ Remove-DbaServerRoleMember -Login $logins
+
+        Removes all the logins found in C:\logins.txt from mycustomrole custom server-level role on server1.
```
[the remark this answers is not in the document]
Comment (MikeyBronowski) [reply functions/Remove-DbaServerRoleMember.ps1:90]: added

File: functions/Remove-DbaServerRoleMember.ps1 [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        The server-level role(s) to process.
+
+    .PARAMETER Login
+        The login(s) to remove from server-level role(s) specified.
+
+    .PARAMETER Role
+        The role(s) to remove from server-level role(s) specified.
+
+    .PARAMETER InputObject
+        Enables piped input from Get-DbaServerRole or New-DbaServerRole
```
[the remark this answers is not in the document]
Comment (MikeyBronowski) [reply functions/Remove-DbaServerRoleMember.ps1]: Valid point. Removed.

File: functions/Remove-DbaServerRoleMember.ps1 (hunk 1 above)
Comment (andreasjordan) [reply functions/Remove-DbaServerRoleMember.ps1:98]: But piping in a server role object, this object would not be bound to $ServerRole, but to $InputObject. So I would remove the `[parameter(ValueFromPipeline)]`. And yes, I think it should be changed in Add-DbaServerRoleMember as well.

File: functions/Remove-DbaServerRoleMember.ps1 (hunk 2 above)
Comment (andreasjordan) [reply functions/Remove-DbaServerRoleMember.ps1:108]: First of all: What does the if tests? It is true if there is no SqlInstance, no ServerRole and no Login. As soon as you provide two of the three, it is false and the function does not stop. Is this intended? I think it is even wrong in Add-DbaServerRoleMember. 
So I would just delete this test. And then maybe think of new tests. And then start with comments explaining what to test in plain english. And then code it.

File: functions/Remove-DbaServerRoleMember.ps1 (hunk 1 above)
Comment (MikeyBronowski) [reply functions/Remove-DbaServerRoleMember.ps1:97]: Got it. Removed from the Remove- function.
